### ISSUE TITLE
Prediction with pred_contrib happened Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by PhungVanDuy** [Microsoft/LightGBM#1338]
## Environment info
Operating System: Ubuntu server 16.04
C++/Python/R version: python 2.7.13
LightGBM: 2.0.12
## Error Message:
Segmentation fault (core dumped)
## Reproducible examples
I performed multiclassification with 54 class after that I want to predict in test set with pred_contrib. But I got this error: "Segmentation fault (core dumped)". Although it still working with pred_contrib = False.
## Steps to reproduce
Here my code, 
```python
def training(X_train, y_train, X_test, y_test, feats):
       params = {
               'task': 'train',
               'boosting_type': 'gbdt',
               'objective': 'multiclass',
               'metric': 'multi_logloss',
               'num_leaves': 31,
               'min_data_in_leaf': 20,
               'learning_rate': 0.05,
               'feature_fraction': 1,
               'bagging_fraction': 1,
               'bagging_freq': 0,
               'num_class': 54,
               'num_iterations':500,
               'verbose': 1,
               'num_threads':25 
       }
       lgb_train = lgb.Dataset(X_train, y_train, free_raw_data=False)
       model = lgb.train(params, lgb_train, valid_sets=lgb_train, feature_name=feats)
       y_pred = model.predict(X_test)
       print model.predict(X_test ,pred_contrib = True)
```
Dataset: Contain 9662 sample belong to 54 class.

**Comment by guolinke** [comment]
can you try to use fewer samples? or use fewer `num_iteration` in predict ?

**Comment by guolinke** [comment]
can you share the model files?  or can you check: are there many trees with only single node (num_leaves=1)?

**Comment by guolinke** [comment]
it would be better if you can send me (guolin.ke@outlook.com) the model file, with a sample of prediction data (csv or lightsvm format).

**Comment by PhungVanDuy** [comment]
@guolinke 
I just sent my model and test data for you. Please check your email. Thanks!

**Comment by guolinke** [comment]
@PhungVanDuy please try the latest master branch. BTW, the n_preds equals to num_row x num_iterations x num_class, please make sure that the length cannot exceed the range of int64.

**Comment by PhungVanDuy** [comment]
@guolinke thanks for your support. I will try it again.
